src/target/riscv/riscv-013.c

```diff
@@ -3114,20 +3132,19 @@ static int modify_privilege(struct target *target, uint64_t *mstatus, uint64_t *
 	return ERROR_OK;
 }
 
-static int read_memory_bus_v0(struct target *target, target_addr_t address,
-		uint32_t size, uint32_t count, uint8_t *buffer, uint32_t increment)
+static int read_memory_bus_v0(struct target *target, riscv_mem_access_info_t access)
 {
-	if (size != increment) {
+	if (access.element_size != access.increment) {
 		LOG_TARGET_ERROR(target, "sba v0 reads only support size==increment");
 		return ERROR_NOT_IMPLEMENTED;
 	}
 
 	LOG_TARGET_DEBUG(target, "System Bus Access: size: %d\tcount:%d\tstart address: 0x%08"
-			TARGET_PRIxADDR, size, count, address);
-	uint8_t *t_buffer = buffer;
-	riscv_addr_t cur_addr = address;
-	riscv_addr_t fin_addr = address + (count * size);
-	uint32_t access = 0;
+			TARGET_PRIxADDR, access.element_size, access.count, access.target_address);
+	uint8_t *t_buffer = access.buffer.read;
+	riscv_addr_t cur_addr = access.target_address;
+	riscv_addr_t fin_addr = access.target_address + (access.count * access.element_size);
+	uint32_t bus_access = 0;
```

Comment:
If you end up in position when you still need to rename this variable, I'd call it `sbcs` (to match the name of the DM register):

```suggestion
	uint32_t sbcs = 0;
```

Reply:
Addressed